src/CarbonAware.CLI/src/Program.cs

```diff
@@ -15,13 +16,22 @@
     .UseCarbonAwareDefaults()
     .Build();
 
-var serviceProvider = new ServiceCollection()
+var builder = new ServiceCollection()
     .AddSingleton<IConfiguration>(config)
     .Configure<CarbonAwareVariablesConfiguration>(
         config.GetSection(CarbonAwareVariablesConfiguration.Key))
-    .AddCarbonAwareEmissionServices(config)
-    .AddLogging(builder => builder.AddDebug())
-    .BuildServiceProvider();
+    .AddLogging(builder => builder.AddDebug());
+
+string? errorMessage = "";
```

Comment:
Why an empty error message? Is it required for the `TryAdd`?

Reply:
Probably because when we first wrote this out the type was `string`, but it was updated to be `string?` for the `TryAdd` function.  Could be fine to leave it uninitialized as long as Line33 `_logger?.LogError(errorMessage)` doesn't complain.